### PR TITLE
feat: step transition without input param

### PR DESCRIPTION
**Pull request by aludwiko**
References https://github.com/lightbend/kalix-proxy/issues/1913

### CODE REVIEW

**Review by efgpinto — APPROVED** [review]
LGTM

File: sdk/java-sdk/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java [hunk 1]
```diff
@@ -383,17 +421,38 @@ public <Input, Output> AsyncCallStepBuilder<Input, Output> asyncCall(Function<In
...
       public CallStepBuilder(String name, Function<Input, DeferredCall<DefCallInput, DefCallOutput>> callFunc) {
         this.name = name;
         this.callFunc = callFunc;
+        this.callSupplier = null;
+      }
+
+      public CallStepBuilder(String name, Supplier<DeferredCall<DefCallInput, DefCallOutput>> callSupplier) {
+        this.name = name;
+        this.callFunc = null;
+        this.callSupplier = callSupplier;
```
[the remark this answers is not in the document]
Comment (johanandren) [reply sdk/java-sdk/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java:455]: Not sure if javac would agree to lambdaize it but something like a marker supertype ("Not for direct user extension") with two more concrete SAM interfaces that is like function and producer, then pattern match internally and call the right method depending on type?

[the remark this answers is not in the document]
Comment (pvlugter) [reply sdk/java-sdk/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java:455]: To only have a function, using `Void` for input type and passing it `null` should work fine for the supplier case. Or could accept `Object` which can then be given anything at all.

[the remark this answers is not in the document]
Comment (aludwiko) [reply sdk/java-sdk/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java]: Ok, thanks for the suggestions, I've tried the simplest one, so `null` :) you can check the outcome.

**Review by johanandren — APPROVED** [review]

**Pull request merged**